app/controllers/application_controller.rb

```diff
@@ -12,6 +13,17 @@ def authenticate_admin!
     end
   end
 
+  ##
+  # Forbid the request if the back_uri parameter is being set to create
+  # offsite links.  The parameter is used in various views throughout the
+  # application.
+  def sanitize_back_uri!
+    if params[:back_uri].present?
+      ok = request.protocol + request.host
+      render nothing: true, status: 403 if !params[:back_uri].start_with? ok
```

Comment:
I want to test it some more in staging, but I decided on a 403 after considering redirects. My understanding is that there is no benign activity that would trigger this response; all `back_uri` params that I've come across so far in normal browsing start with the legitimate scheme plus hostname. In the event of a malicious request, I wanted to avoid wasting any more resources rendering something like the home page or any other page that renders views and makes further API requests.  A 403 very lightweight.